### PR TITLE
new role: user-generate

### PR DESCRIPTION
**Pull request by opadron**
Adds a new role, `user-generate`, that serves as a utility role for creating new users.  It works like the `user` ansible module, but does a few extra things, too.  More details in [roles/user-generate/README.md](https://github.com/opadron/gobig/blob/244b87276250039628245f9bee7d97f0b1b4de5f/roles/user-generate/README.md)

Also adds a few minor fixes.

### CODE REVIEW

File: roles/user-generate/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+The `user-generate` role works like the `user` ansible module, except that it
+will also create a new group for the user if the desired group is not present,
+and add the new user's ssh key to its own list of authorized keys.
+
+#### Variables
+
+|Name        |Default    |Description                                |
+|:-----------|:---------:|:------------------------------------------|
+|name        |(required) |name of the new user                       |
+|crypt_pass  |(generated)|hash of the password to use for the user   |
```
Comment (cpatrick) [inline roles/user-generate/README.md]: I believe this has to be SHA-512. You should note this in the docs.

**Comment by cpatrick** [comment]
One nit, otherwise LGTM.

**Comment by opadron** [comment]
[New README url](https://github.com/opadron/gobig/blob/51a2414ef8aa238cdf07590b74599739234dda76/roles/user-generate/README.md)

Ready for final review.

**Comment by cpatrick** [comment]
LGTM.

**Pull request merged**